pages/Messages/Validation.md

```diff
@@ -0,0 +1,117 @@
+---
+name: Validation
+route: /Messages/Validation
+menu: Messages
+---
+
+# Message Validation
+
+All messages must be validated by announcers before being included in a batch file, however clients and indexers should not rely on announcer validation alone and should perform their own validation on incoming messages as well.
+Message validation in this document is defined as a collection of independent checks which may be run in parallel depending on implementation, however all checks are required to pass for a message to be considered valid.
+
+In general, all implementations must validate announcements for correctness against the [DSNP message schema](/Messages/Overview) and authenticity using [their provided signatures](/Messages/Signatures).
+Client and indexer implementations must also validate the activity pub content of applicable messages for correctness with the [Activity Pub schema](https://www.w3.org/TR/activitypub/) and authenticity using the provided content hash.
+Announcers may choose to skip content validation checks in the interest of performance given the high cost of fetching content.
+
+## Specification Status
+
+| Version | Status |
+---------- | ---------
+| 0.1     | Tentative |
+
+## Purpose
+1. Describe the process of validating messages for inclusion in a batch
+
+## Assumptions
+* All assumptions from [DSNP Messages](/Messages/Overview)
+* All assumptions from [DSNP Signatures](/Messages/Signatures)
+
+## Announcement Correctness
+
+Validation of announcement correctness will vary depending on which fields are present on a particular DSNP message type.
+Validating correctness for a message will consist of verifying that each field in the message is defined and meets the formatting rules listed in the subsections below for the given field type.
+
+#### Addresses
+
+1. Address fields must be exactly 20 bytes.
+
+#### Emoji
+
+1. Emoji fields must not be empty.
+1. Emoji fields must only consist of [Unicode points](https://unicode.org/standard/standard.html) from `U+2000` to `U+2BFF`, from `U+E000` to `U+FFFF`, or from `U+1F000` to `U+10FFFF`.
+
+For example, all of the following should be considered valid emoji:
+
+```
+"😀", "🤌🏼", "👩🏻‍🎤", "🧑🏿‍🏫", "🏳️‍🌈", "🏳️‍⚧️", "⚛︎", "🃑", "♻︎"
+```
+
+Additionally, none of the following should be considered valid:
+
+```
+"F", ":custom-emoji:", "<custom-emoji>", "ᚱ", "ᘐ", "״"
+```
+
+#### Hashes
+
+1. Hash fields must be exactly 32 bytes.
```

Comment:
Since the hash is a byte value, I don't think we need to specify hexadecimal. As long as the column in the batch file takes raw bytes, users ought to be able to pass around hashes in whatever format they want, hex, base64, base53, etc.

Reply:
Hmm... I guess it matters for the signature serialization, but you're right, it doesn't matter as long as parquet is storing bytes not strings. I glanced quickly and it would just be a byte array, so I think we are fine.